ui/Modal/Org/ConfigureEns.svelte

```diff
@@ -44,56 +42,47 @@
         `.${ensResolver.DOMAIN}`,
         ""
       );
-      return { type: "enterEnsName", currentName };
+      return { type: "registerName", currentName };
     } else {
       return { type: "intro" };
     }
   })();
 
-  function bindRegistrationDone(name: string) {
-    const domain = `${name}.${ensResolver.DOMAIN}`;
-    return async function () {
-      // TODO handle exception
-      const registration = await ensResolver.getRegistration(domain);
+  async function registrationDone(result: registerName.Result) {
+    let registration: ensResolver.Registration | null;
+
+    if (result.registration) {
+      registration = result.registration;
+    } else {
+      const domain = `${result.name}.${ensResolver.DOMAIN}`;
+      registration = await ensResolver.getRegistration(domain);
```

Comment:
Could you retain the comment that we need to handle exceptions?

Reply:
In what normal use-case would we arrive here and not have a registration? Should this not be impossible?

Reply:
You’re right, this should never be `null` and we handle this case. But this is different from `getRegistration` throwing an exception. This can happen if the networking fails. In that case we should ensure that the UI is in a good state and the user is informed accordingly.

Reply:
d174180b2b32339dd0ff4fd0dddc055267c76130

Reply:
I meant to restore the comment above the call to `getRegistration` because this is where the exception can happen. I don’t think we need to handle the case where `registration` is not defined. (This should indeed never happen.)

Reply:
a0b03cb4e4f8bc62505848a3bff061095bd1451f